src/TimeoutMigrationTool/MigrationRunner.cs

```diff
@@ -17,42 +17,65 @@ public MigrationRunner(ITimeoutStorage timeoutStorage, ICreateTransportTimeouts
         public async Task Run(IDictionary<string, string> runParameters)
         {
             var forceMigration = runParameters.ContainsKey(ApplicationOptions.ForceMigration);
-            if (forceMigration) 
+            if (forceMigration)
             {
                 await Console.Out.WriteAsync("Migration will be forced.").ConfigureAwait(false);
-                //TODO: is this approach any better?
-                //await timeoutStorage.ResetState().ConfigureAwait(false);
+                await timeoutStorage.Reset().ConfigureAwait(false);
+                await Console.Out.WriteLineAsync("Timeouts storage migration status reset completed.").ConfigureAwait(false);
             }
 
             await Console.Out.WriteAsync("Checking for existing tool state").ConfigureAwait(false);
             var toolState = await timeoutStorage.GetToolState().ConfigureAwait(false);
-            await Console.Out.WriteLineAsync(" - done").ConfigureAwait(false);  
 
-            if (toolState == null || toolState.Status == MigrationStatus.Completed || forceMigration)
+            if (toolState != null)
             {
-                toolState = new ToolState(runParameters);
-                await timeoutStorage.StoreToolState(toolState).ConfigureAwait(false);
+                await Console.Out.WriteLineAsync("Found existing tool state.").ConfigureAwait(false);
+                await Console.Out.WriteLineAsync("Checking migration status.").ConfigureAwait(false);
+
+                if (toolState.Status == MigrationStatus.Completed)
+                {
+                    await Console.Out.WriteLineAsync("Migration already completed. To rerun a migration on this timeout storage run the tool using --forceMigration to reset the migration state.").ConfigureAwait(false);
```

Comment:
I think we should just allow this to continue since the previous run was completed